src/Customer-repo.js

```diff
@@ -36,9 +40,43 @@ class CustomerRepo {
     })
   }
 
+  getRoomsAvailableForGivenDate(date) {
+    this.choosenDate = date;
+    let bookedRooms = []
 
+    this.bookings.forEach(booking => {
+      if(booking.date === date) {
+        bookedRooms.push(booking.roomNumber)
+      }
+    });
+
+  let bookableRooms = this.allRooms.reduce((availableRooms, room)=> {
 
+     if(!bookedRooms.includes(room.number)) {
+       availableRooms.push(room)
+     }
 
+     return availableRooms
+   },[]);
+    // console.log("bookedRooms", bookedRooms);
+    // console.log("BOOKABLE ROOMS", bookableRooms);
+    this.bookableRooms = bookableRooms;
+    domUpdates.showAvailableRooms(bookableRooms)
```

Comment:
Ahh, so this is how you are supposed to update the DOM from the classes. That has been confusing for me so good to see!

Reply:
@pernstrong yeah this is the way to do it. The crappy part about this is that you have to write your code from the get-go with the intention to do this or it will be a pain in the rear to refactor it later. but this makes it relatively easy to spy test.